tendrl/monitoring_integration/sync/dashbaord_sync.py

```diff
@@ -47,120 +59,135 @@ def refresh_dashboard(self):
+        if resources and resource_json:
+            # update dashboard
+            rows = resource_json["dashboard"]["rows"]
+            most_recent_panel_id = rows[-1]["panels"][-1]["id"]
+            resource_json = alert_dashboard.add_panel(
+                resources,
+                resource_type,
+                resource_json,
+                most_recent_panel_id
+            )
+        return resource_json
 
-    def create_gluster_resource(self, resources, resource_type,
-                                integration_id, sds_name):
-        for resource in resources:
-            try:
-                if resource_type == "volumes":
-                    resource_name = str(resource["name"])
-                elif resource_type == "hosts":
-                    resource_name = str(resource["fqdn"]).replace(".", "_")
-                elif resource_type == "bricks":
-                    resource_name = "%s|%s:%s" % (
-                        str(resource["vol_name"]),
-                        resource["hostname"],
-                        resource["brick_path"].replace("|", "/")
-                    )
-            except KeyError:
-                logger.log("debug", NS.get("publisher_id", None),
-                           {'message': "Failed to get resource {} "
-                            "details".format(resource)})
-                continue
-            response = alert_dashboard.add_panel(
-                integration_id,
+    def update_dashboard(self, resource_type, resources, resource_json):
+        no_changes_in_dashboard = False
+        new_resources, alert_dashboard_json, most_recent_panel_id = \
+            alert_dashboard.check_duplicate(
+                copy.deepcopy(resource_json),
+                resources,
+                resource_type
+            )
+        if new_resources and len(
+            alert_dashboard_json["dashboard"]["rows"]
+        ) > 0:
+            alert_dashboard_json = alert_dashboard.add_panel(
+                new_resources,
                 resource_type,
-                resource_name,
-                sds_name
+                alert_dashboard_json,
+                most_recent_panel_id
             )
-            if response:
-                self.log_message(response, resource_name, resource_type)
+        elif len(resource_json["dashboard"]["rows"]) == \
+                len(alert_dashboard_json["dashboard"]["rows"]):
+            no_changes_in_dashboard = True
+        return alert_dashboard_json, no_changes_in_dashboard
+
+    def get_managed_clusters_integration_id(self):
+        managed_clusters_integration_id = []
+        integration_ids = utils.get_resource_keys("", "clusters")
+        for integration_id in integration_ids:
+            cluster_key = "/clusters/%s" % integration_id
+            cluster_is_managed = etcd_utils.read(
+                cluster_key + "/is_managed"
+            ).value
+            if cluster_is_managed.lower() == "yes":
```

Comment:
`If cluster_is_managed in ['yes', 'Yes', 'YES']`

Reply:
i will change this